students/adam_strong/lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+""" Donor classes to be run by cli_main.py
+"""
+
+# Starting data structure of donors:
+
+import copy
+import pathlib
+
+donor_db = {"Scrooge McDuck": [8000.00, 70000.00],
+            "Montgomery Burns": [49.53],
+            "Richie Rich": [1000000.00, 500000.00],
+            "Chet Worthington": [200.00, 44387.63, 10200.00],
+            "Silas Skinflint": [0.25, 1.00, 0.43]}
+
+report_head = '{:20}| {:>15}|{:>15}| {:>15}'.format(
+    'Donor Name', 'Total Given', 'Num Gifts', 'Average Gift')
+
+letter = "\n".join(("", "Dear {},", "",
+                        "    Thank you for your very kind donations totaling ${:.2f}.", "",
+                        "    It will be put to very good use.", "",
+                        "               Sincerely,",
+                        "                  - The team"))
+
+break_line = '-'*72
+
+ty_message = "\n".join(("", "Dear {}",
+                            "Thank you for your generous donation of {:.2f}", ""))
+
+repform = '{:20} ${:>15.2f}  {:>15}  ${:15.2f}'
+
+
+class DonorCollections(object):
+    '''DonorCollections class used to store and manipultate the main dict'''
+
+    donors = copy.deepcopy(donor_db)
+
+    def __init__(self, donors=donors):
+        self.donors = donors
+
+    def add_donor(self, tyname, amt):
+        '''Adds a new donor or updates an existing donor, returns thank you letter'''
+        tyname = tyname.title()
+        if self.donors.get(tyname) is None:
+            self.donors[tyname] = [amt]
+        else:
+            self.donors.get(tyname).append(amt)
+        ty_output = ty_message.format(tyname, amt)
+        return ty_output
```

Comment:
so the idea is to use Donor class as part of your DonorCollection class, which would actually reference Donor instances.

For example, your structure could be a dictionary with key for donor name and value Donor instance.